app/ide-desktop/lib/dashboard/src/components/AriaComponents/Form/Form.tsx

```diff
@@ -83,41 +96,89 @@ export const Form = React.forwardRef(function Form<
   // There is no way to avoid type casting here
   // eslint-disable-next-line @typescript-eslint/no-explicit-any,no-restricted-syntax,@typescript-eslint/no-unsafe-argument
   const formOnSubmit = innerForm.handleSubmit(formMutation.mutateAsync as any)
-  const { formState, clearErrors, getValues, setValue, setError, register, unregister } = innerForm
-
-  const formStateRenderProps: types.FormStateRenderProps<TFieldValues> = {
+  const {
     formState,
+    clearErrors,
+    getValues,
+    setValue,
+    setError,
     register,
     unregister,
+    setFocus,
+    reset,
+  } = innerForm
+
+  const formStateRenderProps: types.FormStateRenderProps<Schema, TFieldValues> = {
+    formState,
+    register: (name, options) => {
+      const registered = register(name, options)
+
+      const onChange: types.UseFormRegisterReturn<Schema, TFieldValues>['onChange'] = value => {
+        if (typeof value === 'object' && value != null && 'target' in value && 'type' in value) {
+          return registered.onChange(value)
+        } else {
+          return registered.onChange({ target: { event: value } })
+        }
+      }
+
+      const result: types.UseFormRegisterReturn<Schema, TFieldValues, typeof name> = {
+        ...registered,
+        ...(registered.disabled != null ? { isDisabled: registered.disabled } : {}),
```

Comment:
is there a reason we don't use `isDisabled: registered.disabled ?? false` here?

Reply:
Mostly because we want to keep it undefined if it is undefined. `IsDisable` reflects the `disabled` prop (mostly for react-aria-components)